mewarpx/mewarpx/utils_store/init_restart_util.py

```diff
@@ -142,7 +142,21 @@ def run_restart(checkpoint_directory="diags",
             )
             return False, None, None
 
-    checkpoint = checkpoints[-1]
+    # check if the checkpoint was completely written by testing if a flux diag
```

Comment:
Is there a case where the checkpoint file could be saved if it was on a diag step and we could copy over the flux diag file from diags to the checkpoint? I'd just like to avoid automatically deleting files unless the file is absolutely unusable.

Also I think it would make more sense for this check to be put earlier, when the program checks for valid checkpoints, and errors out if none are found. The check could also print a warning telling the user that no fluxdiag was found for this checkpoint file, and that it was using an earlier checkpoint instead.

Reply:
On the first point: We have no way of knowing (that I'm aware of) whether the WarpX checkpoint is valid until we attempt a restart and see WarpX fail (usually with something like `terminate called after throwing an instance of 'std::__ios_failure'`). My opinion is that we are better of playing it safe - only attempt restarts from files we know are definitely good - rather than risking having jobs fail. If there was a good way to test the validity of a checkpoint I would agree with you.

On the second point: Good idea, I'll move the check into `get_sorted_checkpoints`.

Reply:
That makes sense. In that case, what do you think about tying the deletion of the checkpoint files with the `clear_old_checkpoints` flag, so automatic removal of checkpoints can be turned on or off at the user's discretion?

Reply:
At least move to a corrupted subdir, so let's just leave it as deleted by default: And we can add this later if necessary for debugging.

Reply:
@KZhu-ME I decided not to move the new functionality in `get_sorted_checkpoints` since that way I can reuse it for `_clean_corrupt_checkpoints`.